Update submodules

MaartenBent opened the pull request:
Test updating expat, jpeg, png and zlib submodules. I'll try to add some more.

MaartenBent committed:
Update submodules

MaartenBent committed:
Add new files and regenerate projects

MaartenBent committed:
Update copyright year

This file was not updated with all the other samples.

MaartenBent commented:
Before merging this, maybe we can also add a libwwebp submodule?
I was testing #24479 and added support for a builtin (CMake) version.

vadz reviewed (approved):
Thanks again for your work on this!

I'd prefer to merge this already and deal with libwebp later in a separate commit, but if you really want to include it in this one, please let me know.

MaartenBent commented:
Fixed the copy-paste mistake in the instructions, and added a line about updating the Xcode projects.

Lets merge this now then and do the webp stuff some other time.
